### PR TITLE
add Task-Create

### PR DESCRIPTION
**Pull request by emelbaer**
Thanks for the api!

adding Task-Create :)

### CODE REVIEW

**Comment by emelbaer** [comment]
Any future plans?

**Comment by N4S4** [comment]
Thank you for contributing adding 'Task-Create'!!
there are few little issues in your code which I will comment in each file of the pull request.

have you tested your code yet?

Future plans are in progress, thanks to @v7x great work, already with a completely changed way of how the api works, the active branch is [THIS](https://github.com/N4S4/synology-api/tree/Improving-code) if you want to have a look you are more than welcome!
Let me know what do you think about!

File: synology_api/auth.py [hunk 1]
```diff
@@ -92,28 +92,21 @@ def request_data(self, api_name, api_path, req_param, method=None, response_json
...
-        if method is None:
-            method = 'get'
-
         req_param['_sid'] = self._sid
 
-        if method is 'get':
-            url = ('%s%s' % (self._base_url, api_path)) + '?api=' + api_name
+        url = ('%s%s' % (self._base_url, api_path)) + '?api=' + api_name
+        # checking and handling HTTP-Method (perform a request)
+        if method.lower() is 'get' or method is None:
```
Comment (N4S4) [inline synology_api/auth.py]: Default  'method' must be specified as all function do not specify what method will be used when 'request_data()' is called, 'None' will pass through the conditional rising errors like 'NoneType' object has no attribute 'lower'

File: setup.py [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
     packages=find_packages(exclude=['tests*']),
     license='MIT',
     description='Python Synology API Wrapper',
     long_description=open('README.md').read(),
     install_requires=['requests'],
-    url='https://github.com/N4S4/synology-api',
-    author='Renato Visaggio',
-    author_email='renatovisaggio@gmail.com'
+    url='https://github.com/emelbaer/synology-api',
+    author='Emel B (c) Renato Visaggio',
```
Comment (N4S4) [inline setup.py]: The following are just little requests as I spent a considerable amount of time reading the docs and writing this wrapper even if is not perfect at all, 

I will soon write some contributing guide to the package, for now I will write just a comment on how I would like it to be done if is requested to merge to the master branch;


- I would keep major version changes for when the api wrapper is complete in all its functions. 
- url has to be referred to the master branch

- Author email has to stay, if you wish you can add yours to the list separated by comma and after mine

- It is fine for me to add authors to the list but the only thing I ask is to do it after my name separated by commas, 
  but my plan was to add names myself of anyone giving major contribution spending time on it like you.

do not take it personal this are the future contributions rules to merge to the master branch which I will add to the instructions once ill have a bit of time

**Comment by v7x** [comment]
@emelbaer downloadstation.create_task() should be rewritten to be in the new style, which you can see in the "Improving Code" branch. I don't have it very well documented at the moment, so if you need help understanding what's going on when you add a new method let me know. But basically you can just a different method in both the old and new branch to see what it looks like.

**Comment by tduboys** [comment]
what appends if destination does not exist ?
is there an error during task creation or just the task will fail after being created ?

**Comment by nicholaschum** [comment]
@v7x I have the updated code on my fork:

https://github.com/nicholaschum/synology-api/blob/Improving-code/synology_api/downloadstation.py#L67-L88

**Comment by N4S4** [comment]
@nicholaschum yes if testing was succesful you might PR for final test and merge

Thank you very much

**Pull request closed without merging**